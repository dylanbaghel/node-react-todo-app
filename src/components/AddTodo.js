import React from 'react';
import { connect } from 'react-redux';

import TodoForm from './TodoForm';
import { startAddTodo } from './../actions/todosActions';

export const AddTodo = (props) => {
    return (
        <div>
            <TodoForm 
                onFormSubmit={(todo) => {
                    props.startAddTodo(todo.text);
                }}
            />
        </div>
    );
};

const mapDispatchToProps = (dispatch) => {
    return {
        startAddTodo: (text) => dispatch(startAddTodo(text))
    };
};

export default connect(undefined, mapDispatchToProps)(AddTodo);