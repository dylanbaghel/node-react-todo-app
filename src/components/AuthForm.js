import React from 'react';
import classnames from 'classnames';
import isEmail from 'validator/lib/isEmail';

class AuthForm extends React.Component {
    constructor(props) {
        super(props);
        this.handleEmailChange = this.handleEmailChange.bind(this);
        this.handlePasswordChange = this.handlePasswordChange.bind(this);
        this.onFormSubmit = this.onFormSubmit.bind(this);
        this.state = {
            email: '',
            password: '',
            errors: {}
        };
    }

    handleEmailChange(e) {
        const email = e.target.value;
        this.setState(() => ({
            email
        }));
    }

    handlePasswordChange(e) {
        const password = e.target.value;
        this.setState(() => ({
            password
        }));
    }

    onFormSubmit(e) {
        e.preventDefault();

        if (!this.state.email.trim()) {
            this.setState(() => ({
                errors: { email: 'Please Enter Email' }
            }));
            return;
        } else if (!isEmail(this.state.email)) {
            this.setState(() => ({
                errors: {email: 'Please Enter Valid Email'}
            }));
            return;
        }
        if (!this.state.password.trim()) {
            this.setState(() => ({
                errors: { password: 'Please Enter Password' }
            }));
            return;
        } else if (this.state.password.length < 6) {
            this.setState(() => ({
                errors: { password: 'Password Must Be Greater Then 6 Characters.'}
            }));
            return;
        }
        this.setState(() => ({
            errors: {}
        }));
        this.props.onSubmit({
            email: this.state.email,
            password: this.state.password
        });
    }

    render() {
        return (
            <div>
                <form onSubmit={this.onFormSubmit}>
                    <div className="form-group">
                        <input
                            type="text"
                            placeholder="Email"
                            className={classnames('form-control', {
                                'is-invalid': this.state.errors.email
                            })}
                            value={this.state.email}
                            onChange={this.handleEmailChange}
                            onKeyUp={(e) => {
                                const email = e.target.value;
                                if (!isEmail(email)) {
                                    this.setState(() => ({
                                        errors: {email: 'Please Enter Valid Email'}
                                    }));
                                } else {
                                    this.setState(() => ({
                                        errors: {}
                                    }));
                                }
                            }}
                        />
                        <div className="invalid-feedback">
                            <small className="text-danger">{this.state.errors.email}</small>
                        </div>
                    </div>
                    <div className="form-group">
                        <input
                            type="password"
                            placeholder="Password"
                            className={classnames('form-control', {
                                'is-invalid': this.state.errors.password
                            })}
                            value={this.state.password}
                            onChange={this.handlePasswordChange}
                        />
                        <div className="invalid-feedback">
                            <small className="text-danger">{this.state.errors.password}</small>
                        </div>
                    </div>
                    <button className="btn btn-primary">Submit</button>
                </form>
            </div>
        );
    }
}

export default AuthForm;