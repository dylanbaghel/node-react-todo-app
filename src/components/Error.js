import React from 'react';
import { connect } from 'react-redux';

export const Error = (props) => {
    return (
        <div className="alert alert-danger">
            {props.error.error_text}
        </div>
    );
};

const mapStateToProps = (state) => {
    return {
        error: state.error
    };
};

export default connect(mapStateToProps)(Error);