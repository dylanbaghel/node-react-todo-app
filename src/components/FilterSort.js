import React from 'react';
import { connect } from 'react-redux';

import { setTextFilter, sortByASC, sortByDSC } from './../actions/filterAction';

const FilterSort = (props) => {
    return (
        <div className="d-flex justify-content-between">
            <div className="form-group w-75">
                <input
                    type="text"
                    placeholder="Search Todo...."
                    onChange={(e) => {
                        let text = e.target.value;
                        props.setTextFilter(text.trim());
                    }}
                    className="form-control"
                />
            </div>
            <div className="form-group">
                <select className="form-control" onChange={(e) => {
                    if (e.target.value === 'asc') {
                        props.sortByASC();
                    } else if (e.target.value === 'dsc') {
                        props.sortByDSC();
                    }
                }}>
                    <option value="asc">A-Z</option>
                    <option value="dsc">Z-A</option>
                </select>
            </div>
        </div>
    );
};

const mapDispatchToProps = (dispatch) => {
    return {
        setTextFilter: (text) => dispatch(setTextFilter(text)),
        sortByASC: () => dispatch(sortByASC()),
        sortByDSC: () => dispatch(sortByDSC())
    };
};

export default connect(undefined, mapDispatchToProps)(FilterSort);