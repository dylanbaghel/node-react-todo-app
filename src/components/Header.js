import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { logoutUser } from './../actions/authAction';

const Header = (props) => {
    const publicLinks = (
        <ul className="navbar-nav ml-auto">
            <li className="nav-item">
                <Link
                    to="/login"
                    className="nav-link"
                >
                    Login
                </Link>
            </li>
            <li className="nav-item">
                <Link
                    to="/signup"
                    className="nav-link"
                >
                    Sign Up
                </Link>
            </li>
        </ul>
    );
    const privateLinks = (
        <ul className="navbar-nav ml-auto">
            <li className="nav-item">
                <Link
                    to="/"
                    className="nav-link"
                    onClick={() => {
                        props.logoutUser();
                    }}
                >
                    Logout
            </Link>
            </li>
        </ul>
    );
    return (
        <nav className="navbar navbar-expand navbar-dark bg-dark mb-5">
            <div className="container">
                <Link to="/" className="navbar-brand">Todo</Link>
                {
                    props.isAuthenticated ? privateLinks : publicLinks
                }
            </div>
        </nav>
    );
};

const mapStateToProps = (state) => {
    return {
        isAuthenticated: !!state.auth.uid
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        logoutUser: () => dispatch(logoutUser())
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);