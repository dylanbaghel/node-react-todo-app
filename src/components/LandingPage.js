import React from 'react';
import { Link } from 'react-router-dom';

const LandingPage = () => {
    return (
        <div className="container text-center d-flex flex-column justify-content-center landing">
            <h1>Welcome To Todo App</h1>
            <p>Created By Abhishek Baghel</p>
            <div className="d-flex justify-content-center">
                <Link to="/login" className="btn btn-primary mr-1">Login</Link>
                <Link to="/signup" className="btn btn-danger ml-1">Sign Up</Link>
            </div>
        </div>
    );
};

export default LandingPage;