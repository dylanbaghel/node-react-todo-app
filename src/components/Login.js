import React from 'react';
import { connect } from 'react-redux';

import AuthForm from './AuthForm';
import { loginUser } from './../actions/authAction';
import Error from './Error';

export const Login = (props) => {
    return (
        <div className="container">
            {props.hasErrors && <Error />}
            <div className="card card-body">
                <h4 className="card-title">Login</h4>
                <AuthForm 
                    onSubmit={(userData) => {
                        props.loginUser(userData);
                    }}
                />
            </div> 
        </div>
    );
};

const mapDispatchToProps = (dispatch) => {
    return {
        loginUser: (userData) => dispatch(loginUser(userData))
    };
};

const mapStateToProps = (state) => {
    return {
        hasErrors: !!state.error.error_text
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);