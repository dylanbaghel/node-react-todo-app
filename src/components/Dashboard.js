import React from 'react';
import { connect } from 'react-redux';

import { startSetTodos } from './../actions/todosActions';
import preLoader from './../gifs/preloader.gif';
import Todo from './Todo';
import AddTodo from './AddTodo';
import FilterSort from './FilterSort';
import { getSelectedTodos } from './../selectors/getSelectedTodos';

class Dashboard extends React.Component {
    constructor(props) {
        super(props);

        this.loadingJSX = (
            <div className="preloader">
                <img src={preLoader} />
            </div>
        );
    }

    componentDidMount() {
        this.props.startSetTodos();
    }

    render() {
        return (
            this.props.isLoading ? this.loadingJSX : <div className="container mx-auto w-75">
            <div className="card card-body mb-3">
                <FilterSort />
            </div>
            <div className="card  mb-1">
                <div className="card-header">Todos</div>
                <div className="card-body">
                    {this.props.todos.length > 0 ?
                        this.props.todos.map((todo) => {
                            return <Todo
                                todo={todo}
                                key={todo._id}
                            />;
                        })
                        : <h1>No Todos Found</h1>
                    }
                </div>
            </div>
            <div className="card card-body mb-5">
                <AddTodo />
            </div>
        </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        todos: getSelectedTodos(state.todos, state.filters),
        isLoading: state.isLoading
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        startSetTodos: () => dispatch(startSetTodos())
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);