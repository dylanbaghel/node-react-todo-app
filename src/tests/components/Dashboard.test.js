import React from 'react';
import { shallow } from 'enzyme';

import { Dashboard } from './../../components/Dashboard';
import todos from './../fixtures/seedData';

let startSetTodos;

beforeEach(() => {
    startSetTodos = jest.fn();
});

test('should render dashboard todo list page with no todos and no loading', () => {
    const wrapper = shallow(<Dashboard todos={[]} isLoading={false} startSetTodos={startSetTodos}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render dashboard todo list page with no todos and loading todos', () => {
    const wrapper = shallow(<Dashboard todos={[]} isLoading={true} startSetTodos={startSetTodos}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render dashboard todo list page with todos and no loading', () => {
    const wrapper = shallow(<Dashboard todos={todos} isLoading={false} startSetTodos={startSetTodos}/>);
    expect(wrapper).toMatchSnapshot();
});
