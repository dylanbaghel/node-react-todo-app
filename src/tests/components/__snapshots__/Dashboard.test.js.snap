// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render dashboard todo list page with no todos and loading todos 1`] = `
<div
  className="preloader"
>
  <img
    alt="Preloader"
    src="preloader.gif"
  />
</div>
`;

exports[`should render dashboard todo list page with no todos and no loading 1`] = `
<div
  className="container mx-auto w-75"
>
  <div
    className="card card-body mb-3"
  >
    <Connect(FilterSort) />
  </div>
  <div
    className="card  mb-1"
  >
    <div
      className="card-header"
    >
      Todos
    </div>
    <div
      className="card-body"
    >
      <h1>
        No Todos Found
      </h1>
    </div>
  </div>
  <div
    className="card card-body mb-5"
  >
    <Connect(AddTodo) />
  </div>
</div>
`;

exports[`should render dashboard todo list page with todos and no loading 1`] = `
<div
  className="container mx-auto w-75"
>
  <div
    className="card card-body mb-3"
  >
    <Connect(FilterSort) />
  </div>
  <div
    className="card  mb-1"
  >
    <div
      className="card-header"
    >
      Todos
    </div>
    <div
      className="card-body"
    >
      <Connect(Todo)
        key="abcd123"
        todo={
          Object {
            "_id": "abcd123",
            "completed": false,
            "completedAt": 0,
            "text": "rent",
          }
        }
      />
      <Connect(Todo)
        key="abcd1234"
        todo={
          Object {
            "_id": "abcd1234",
            "completed": true,
            "completedAt": 0,
            "text": "water bill",
          }
        }
      />
      <Connect(Todo)
        key="abcd12345"
        todo={
          Object {
            "_id": "abcd12345",
            "completed": false,
            "completedAt": 0,
            "text": "gas bill",
          }
        }
      />
    </div>
  </div>
  <div
    className="card card-body mb-5"
  >
    <Connect(AddTodo) />
  </div>
</div>
`;
